src/serac/physics/materials/tests/parameterized_nonlinear_J2_material.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2019-2022, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file nonlinear_J2_material.cpp
+ */
+
+#include "serac/physics/materials/parameterized_solid_material.hpp"
+
+#include <iostream>
+#include <fstream>
+
+#include <gtest/gtest.h>
+
+#include "serac/numerics/functional/tensor.hpp"
+
+namespace serac {
+
+TEST(ParameterizedNonlinearJ2Material, DerivativeWrtYieldStrength)
+{
+  solid_mechanics::ParameterizedJ2Nonlinear        material{.E = 200e6, .nu = 0.25, .density = 1.0};
+  solid_mechanics::ParameterizedJ2Nonlinear::State internal_state{};
+  tensor<double, 3, 3>                             H{
+      {{0.35490513, 0.60419905, 0.4275843}, {0.23061597, 0.6735498, 0.43953657}, {0.25099766, 0.27730572, 0.7678207}}};
+  double sigma_y         = 100e3;
+  double sigma_sat       = 500e3;
+  double strain_constant = 0.1;
+
+  auto internal_state_new = internal_state;
+
+  auto stress = material(internal_state_new, H, make_dual(sigma_y), sigma_sat, strain_constant);
+
+  double h                   = 1e-4 * sigma_y;
+  auto   internal_state_copy = internal_state;
+  auto   stress_p            = material(internal_state_copy, H, sigma_y + h, sigma_sat, strain_constant);
+  internal_state_copy        = internal_state;
+  auto stress_m              = material(internal_state_copy, H, sigma_y - h, sigma_sat, strain_constant);
+  auto finite_difference     = 1. / (2 * h) * (stress_p - stress_m);
+
+  auto error = get_gradient(stress) - finite_difference;
+
+  EXPECT_LT(norm(error) / norm(finite_difference), 1e-5);
+};
+
+TEST(ParameterizedNonlinearJ2Material, DerivativeWrtSaturationStrength)
+{
+  solid_mechanics::ParameterizedJ2Nonlinear        material{.E = 200e6, .nu = 0.25, .density = 1.0};
+  solid_mechanics::ParameterizedJ2Nonlinear::State internal_state{};
+  tensor<double, 3, 3>                             H{
+      {{0.35490513, 0.60419905, 0.4275843}, {0.23061597, 0.6735498, 0.43953657}, {0.25099766, 0.27730572, 0.7678207}}};
+  double sigma_y         = 100e3;
+  double sigma_sat       = 500e3;
+  double strain_constant = 0.1;
+
+  auto internal_state_new = internal_state;
+
+  auto stress = material(internal_state_new, H, sigma_y, make_dual(sigma_sat), strain_constant);
+
+  double h                   = 1e-4 * sigma_sat;
+  auto   internal_state_copy = internal_state;
+  auto   stress_p            = material(internal_state_copy, H, sigma_y, sigma_sat + h, strain_constant);
+  internal_state_copy        = internal_state;
+  auto stress_m              = material(internal_state_copy, H, sigma_y, sigma_sat - h, strain_constant);
+  auto finite_difference     = 1. / (2 * h) * (stress_p - stress_m);
```

Comment:
I trust that this line is correct as written, but I always worry about the order-dependence of these operators
```suggestion
  auto finite_difference     = (1.0 / (2 * h)) * (stress_p - stress_m);
```
or 
```suggestion
  auto finite_difference     = (stress_p - stress_m) / (2.0 * h);
```

Reply:
It does do what we want, since division and multiplication have the same precedence and have left-to-right associativity. I'm used to reading these things, but I have no objections to the extra parentheses from your first example to make the intent clearer. I'll push that change.
The second suggestion is different, in that it will cause 9 floating point divisions (one for each stress component) instead of 9 multiplications. I reflexively avoid FP divides as an old ingrained habit, but you would know better than me if that is still warranted on modern hardware. (This particular use is obviously not performance-critical, but I'd like to know if I  should change my patterns).